core/tests/mod.rs

```diff
@@ -0,0 +1,646 @@
+use std::{
+    collections::{HashMap, HashSet},
+    path::Path,
+    str::FromStr,
+    sync::{Arc, Mutex},
+    time::Instant,
+};
+
+use iroha_config::iroha::Configuration;
+use iroha_core::{
+    block::BlockBuilder,
+    kura::Kura,
+    prelude::*,
+    queue::Queue,
+    smartcontracts::Registrable,
+    sumeragi::{message::*, view_change::*, *},
+};
+use iroha_data_model::{block::*, prelude::*, transaction::Accept};
+use iroha_genesis::{GenesisNetwork, GenesisNetworkTrait, RawGenesisBlock};
+use once_cell::sync::Lazy;
+use rand::{seq::IteratorRandom, SeedableRng};
+
+#[test]
+fn sumeragi_commits_and_broadcasts_genesis() {
+    init_telemetry();
+
+    let pb_hash = Arc::new(Mutex::new(None));
+    let pb_hash2 = pb_hash.clone();
+    /* Can't have 1 peer, then block is not sent. Can't have < 4 peers because non genesis peers get upset. */
+    let (_keypairs, peer_ids, _configs, _queues, sumeragis) = set_up_network(
+        4,
+        Arc::new(move |packet, _peer_id, _sumeragis| {
+            if let message::Message::BlockCreated(block) = packet.message {
+                let vb: VersionedCommittedBlock = block.block.commit_unchecked().into();
+                *pb_hash2.lock().unwrap() = Some(vb.hash());
+            }
+        }),
+    );
+    let sumeragis = sumeragis.lock().unwrap();
+    let sumeragi = sumeragis.get(&peer_ids[0]).unwrap();
+
+    for _ in 0..300 {
```

Comment:
Please, make `300` a constant with meaningful name. Right now it's a magic number

Reply:
Same for other places where it's hard to understand what's going on. I.e. when `_` is used as a variable for iteration

Reply:
Personally don't mind it. Magic numbers especially in tests actually help with readability. 

**However**, What I would like instead, is for this function to be parametric (non-test), and that you could change the number of iterations at the call site inside of a test with a proper name. 

This would later allow us to convert the regular test into a `proptest` and handle unforeseen edgecases.